packages/manager/modules/pci/src/projects/projects.routing.js

```diff
@@ -56,33 +58,27 @@ export default /* @ngInject */ ($stateProvider) => {
       redirectContext: /* @ngInject */ ($transition$) =>
         $transition$.params().context,
 
-      redirectCategory: /* @ngInject */ ($transition$) =>
-        $transition$.params().category,
-
       redirectTarget: /* @ngInject */ ($transition$) =>
-        $transition$.params().target,
+        JSON.parse($transition$.params().target),
 
-      isRedirectRequired: /* @ngInject */ (
-        redirectContext,
-        redirectCategory,
-        redirectTarget,
-      ) => {
-        return !!(redirectContext && redirectCategory && redirectTarget);
+      isRedirectRequired: /* @ngInject */ (redirectTarget) => {
+        const { category, state } = redirectTarget;
+        const isStateExist = has(PCI_FEATURES_STATES, `${category}.${state}`);
```

Comment:
I am not sure duable.
The case that i have to mange is to check if full path is exist.

In your case if `category` is not matched an error will be trigger because access to `state` it not possible.
i am using `has` for this purpose

Reply:
Then : 

```
const isStateExist = PCI_FEATURES_STATES[category] ? !!PCI_FEATURES_STATES[category][state] : false;
```

Reply:
yep i know, but not elegant from my point of view.
that why i use the `has` function.

Thanks